include/circt/Dialect/FIRRTL/FIRRTLTypes.h

```diff
@@ -384,6 +384,9 @@ class FVectorType : public FIRRTLType::TypeBase<FVectorType, FIRRTLType,
   std::pair<unsigned, bool> rootChildFieldID(unsigned fieldID, unsigned index);
 };
 
+// Get the bit width for this type, return None if unknown.
+llvm::Optional<int32_t> getBitWidth(FIRRTLType type);
```

Comment:
Isn't this just getBitWidthOrSentinel?

Reply:
Ah, that sentinels out on aggregates.  Fair question, should it?  Seems redundant to have 2 almost identical things.

Reply:
`getBitWidthOrSentinel` returns `-2` on an aggregate type, whereas this `getBitWidth` can recursively traverse the fields and return the total bitwidth of the leaf elements.